pkg/providers/kubernetes/provider.go

```diff
@@ -17,13 +18,20 @@ import (
 	listers "k8s.io/client-go/listers/core/v1"
 )
 
+// TCPPortMapper is capable of retrieving a TCP port mapping for a given service.
+type TCPPortMapper interface {
```

Comment:
Why is this interface different than for the controller?

Reply:
A provider shouldn't be able to Add a new mapping, it should only be capable of Finding an existing one. This is why we chose to expose only this method.

Reply:
Then should we use the same interface name as the interface that does have other methods?

Reply:
It definitely makes sense. I'm not convinced by `TCPPortMapperReadOnly`. What about `TCPPortFinder`? :)

Reply:
Sounds good.